pkgs/servers/photoprism/frontend.nix

```
@@ -0,0 +1,48 @@
+{ nodePackages, nodejs-14_x, stdenv, src, version, ... }:
+
+let
+  nodeDependencies = nodePackages.photoprism-frontend.override {
+    inherit version;
+    name = "photoprism-frontend-dependencies";
+
+    # Workaround for lack of sourceRoot option in buildNodePackage.
+    src = "${src}/frontend";
+
+    meta.broken = false;
+  };
+in
+stdenv.mkDerivation {
+  inherit src version;
+  name = "photoprism-frontend";
+
+  buildInputs = [ nodejs-14_x ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    pushd frontend
+    ln -s ${nodeDependencies}/lib/node_modules/photoprism/node_modules ./node_modules
+    export PATH="${nodeDependencies}/lib/node_modules/photoprism/node_modules/.bin:$PATH"
+    NODE_ENV=production npm run build
+    popd
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir $out
+    cp -r assets $out/
```

```suggestion
    install -Dt $out assets
```
though I'm not `assets` is the best directory name since it's very nonstandard, but it's okay for now.

---

Maybe `share/photoprism`?

---

Seems like a good one :)

---

Yeah that should work.
What's the default value for this environment variable, i.e. why did $out/assets work?